app/models/manageiq/providers/redhat/infra_manager/event_parser.rb

```diff
@@ -78,7 +78,11 @@ def self.parse_new_vm(vm, message, event_type, ems)
   end
 
   def self.parse_target_name(message, event_type)
-    if %w(NETWORK_ADD_VM_INTERFACE NETWORK_INTERFACE_PLUGGED_INTO_VM).include?(event_type)
+    case event_type
+    when "NETWORK_ADD_VM_INTERFACE"
+      # sample message: "Interface nic1 (VirtIO) was added to VM v5. (User: admin@internal-authz)"
+      message.split(/\s/)[7][0...-1]
```

Comment:
It is a limitation of a model that we use and there are edge cases where it is the only way to understand the name. I agree that we rely on fragile message but there is no other way around it.